### DISCUSSION TITLE
ffmpeg device source in go2rtc not seeing hardware

### DISCUSSION THREAD

**Discussion by finishthepint** [blakeblackshear/frigate#5979] [Ask A Question] [answered]
Hello, I've been using a USB camera for a while now in frigate using "- path: /dev/video0" as the input.  I've switched to 0.12 and am trying to get it setup with go2rtc but i can't seem to get it to work.  I've tried going to the go2rtc dashboard like the go2rtc documentation suggests but i just see "undefined" there.  If i try to use "ffmpeg:device?video=0" as one of my go2rtc streams, it also does not work.  I'm using frigate as an Add-On in a Supervised Home Assistant install and i'm wondering if it has something to do with hardware permission?

**Comment by NickM-27** [comment]
If you're running the addon you'll need to use the full access addon with protected mode set to false for the best chance.

**Reply by finishthepint** [reply]
Thanks for the reply and all your hard work!  I'll give that a try but it's a bit surprising since frigate can get to the device using /dev/video0 so i would think go2rtc would also be able to access it also.

**Reply by NickM-27** [reply]
Ah missed that detail, never mind. What do the go2rtc logs say?

**Reply by finishthepint** [reply]
It looks like i get a lot of these errors:
`2023-04-11 03:39:14.198928481  20:39:14.198 WRN [rtsp] error="source 0 error: exec: exit status 1" stream=Living_Room
2023-04-11 03:46:54.218139628  20:46:54.217 ERR github.com/AlexxIT/go2rtc/cmd/streams/producer.go:53 > error="exec: exit status 1" url=ffmpeg:device?video=0
2023-04-11 03:46:54.232790569  20:46:54.232 ERR github.com/AlexxIT/go2rtc/cmd/streams/producer.go:53 > error="exec: exit status 1" url=ffmpeg:device?video=0
2023-04-11 03:46:54.232845334  20:46:54.232 WRN [rtsp] error="source 0 error: exec: exit status 1" stream=Living_Room`

**Reply by NickM-27** [reply]
You need to update the go2rtc config to have trace level for exec logs

**Reply by finishthepint** [reply, marked as answer]
Finally got time to look into this a little more and got it work.  Here's what worked for me in case others run into this issue.  To enable more logging, changed my go2rtc config to:
`go2rtc:
  log:
    level: debug  # debug level
    api: trace
    exec: debug
    ngrok: info
    rtsp: warn
    streams: error
    webrtc: fatal`

I then saw an error that it couldn't open the device.  This was strange because i had already suspected this and had totally removed the webcam from the normal frigate config and restarted the addon.  Anyway, i decided to remove the config from frigate again and do a full system reboot this time which got the camera working but i ran into a new error: "go2rtc x264 [error]: high profile doesn't support 4:2:2"

A little googling and this is the config that finally worked for me:
`go2rtc:
  streams:
    Living_Room:
      - ffmpeg:device?video=0&resolution=1280x720#video=h264#raw=-pix_fmt yuv420p#hardware`
